src/actions/navigation.js

```diff
@@ -26,18 +27,39 @@ export const chooseDisplayComponentFromURL = (url) => {
 };
 
 /*
- * All the Fetch Promises are created before first render. When trying the cache we `await`.
- * If the Fetch is not finished, this will wait for it to end. Subsequent awaits will immeditaly return the result.
- * For the landing dataset, no problem either because await on a value just returns the value.
+ * `datasets` is populated with `Dataset()` instances for each dataset in a narrative.
+ * Each instance contains promises to represent the main and sidecar datafiles.
+ * If the Fetch is not finished, this will wait for it to end. Subsequent awaits will immediately return the result.
  */
-const tryCacheThenFetch = async (mainTreeName, secondTreeName, state) => {
-  if (state.jsonCache && state.jsonCache.jsons && state.jsonCache.jsons[mainTreeName] !== undefined) {
-    return {
-      json: await state.jsonCache.jsons[mainTreeName],
-      secondJson: await state.jsonCache.jsons[secondTreeName]
-    };
+const updateNarrativeDataset = async (dispatch, datasets, narrativeBlocks, path, query) => {
+  try {
+    const [mainTreeName, secondTreeName] = getDatasetNamesFromUrl(path);
+    const mainDataset = datasets[mainTreeName];
+    const secondDataset = datasets[secondTreeName];
+    const mainJson = await mainDataset.main;
+    const secondJson = secondDataset ? (await secondDataset.main) : false;
+    dispatch({
+      type: URL_QUERY_CHANGE_WITH_COMPUTED_STATE,
+      ...createStateFromQueryOrJSONs({
+        json: mainJson,
+        secondTreeDataset: secondJson,
+        mainTreeName,
+        secondTreeName,
+        narrativeBlocks,
+        query,
+        dispatch
+      }),
+      pushState: true,
+      query
+    });
+    mainDataset.loadSidecars(dispatch);
```

Comment:
Makes sense in terms of frequencies to only fetch for the main dataset and not the second one since it seems like only the main dataset's frequencies can be displayed in a tanglegram view. What about for root sequences? Sorry, I'm not as familiar with how that file works / is displayed.